File: src/test/java/com/api/igdb/TestTwitchToken.java
package com.api.igdb;

import com.api.igdb.request.TwitchAuthenticator;
import com.api.igdb.utils.TwitchToken;
import org.junit.jupiter.api.Test;

import java.time.Duration;

import static org.junit.jupiter.api.Assertions.assertNotNull;
import static org.junit.jupiter.api.Assertions.assertTrue;

public class TestTwitchToken {

    @Test
    void testGetTwitchToken() {
        // Create the Twitch authenticator instance
        TwitchAuthenticator tAuth = TwitchAuthenticator.INSTANCE;
        // request a new Twitch Authentication token
        TwitchToken token = tAuth.requestTwitchToken(System.getenv("client_id"), System.getenv("client_secret"));

        assertNotNull(token);
        assertTrue(token.getExpires_in() > Duration.ofDays(57).toSeconds());
        long tokenFutureDate = System.currentTimeMillis() + Duration.ofDays(57).toSeconds();
        assertTrue(tokenFutureDate > token.getExpiresUnix());
    }
}

File: src/test/kotlin/com/api/igdb/TestTwitchToken.kt
package com.api.igdb

import com.api.igdb.request.TwitchAuthenticator
import org.junit.jupiter.api.Test

class TestTwitchToken {
    @Test
    fun testGetTwitchToken() {
        // Create the Twitch authenticator instance
        val tAuth = TwitchAuthenticator
        // request a new Twitch Authentication token
        val token = tAuth.requestTwitchToken(System.getenv("client_id"), System.getenv("client_secret"))

        assert(token!!.expires_in > 5000000)
        val tokenFutureDate = System.currentTimeMillis() + 5000000 // 57 days
        assert(tokenFutureDate > token.getExpiresUnix())
        assert(token != null)
    }
}